liquibase-maven-plugin/src/main/java/org/liquibase/maven/plugins/LiquibaseSnapshotMojo.java

```diff
@@ -0,0 +1,109 @@
+package org.liquibase.maven.plugins;
+
+import liquibase.Liquibase;
+import liquibase.command.CommandExecutionException;
+import liquibase.command.CommandFactory;
+import liquibase.command.CommandScope;
+import liquibase.command.core.SnapshotCommand;
+import liquibase.command.core.SnapshotCommandStep;
+import liquibase.command.core.helpers.DbUrlConnectionArgumentsCommandStep;
+import liquibase.database.Database;
+import liquibase.exception.LiquibaseException;
+import liquibase.util.StringUtil;
+import org.apache.maven.plugin.MojoExecutionException;
+
+import java.io.*;
+
+/**
+ * <p>The snapshot command captures the current state of the url database, which is the target database.<p/>
+ *
+ * @author Timur Malikin
+ * @goal snapshot
+ */
+public class LiquibaseSnapshotMojo extends AbstractLiquibaseChangeLogMojo {
+
+    /**
+     * Creates a JSON or YAML file that represents the current state of the database.
+     *
+     * @parameter property="liquibase.snapshotFormat" default-value="json"
+     */
+    protected String snapshotFormat;
+
+    /**
+     * Specifies the file path to where the snapshot JSON or YAML will be written.
+     *
+     * @parameter property="liquibase.outputFile"
+     */
+    protected File outputFile;
+
+    /**
+     * The writer for writing the snapshot file.
+     */
+    private Writer outputWriter;
```

Comment:
This one should be a local variable.

Reply:
Thanks, I'll fix this later today and SonarQube issues

Reply:
Thank you, @malikin. Please let me know if you need any help!

Reply:
We don't need it actually, with the "CommandStep framework" code is much simpler. I removed the unused method and this property too, should be good now.

Reply:
Thanks!